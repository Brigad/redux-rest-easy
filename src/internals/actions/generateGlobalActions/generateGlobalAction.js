const generateGlobalAction = actionType => () => ({
  type: actionType,
});

export default generateGlobalAction;
