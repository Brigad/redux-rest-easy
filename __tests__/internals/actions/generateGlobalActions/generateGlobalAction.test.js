import generateGlobalAction from '../../../../src/internals/actions/generateGlobalActions/generateGlobalAction';

const ACTION_TYPE = '@@rest-easy/@global/@reset/RESET_ALL';

describe('generateGlobalAction', () => {
  test('no args', () => {
    expect(generateGlobalAction()()).toMatchSnapshot();
  });

  test('happy path', () => {
    expect(generateGlobalAction(ACTION_TYPE)()).toMatchSnapshot();
  });
});
