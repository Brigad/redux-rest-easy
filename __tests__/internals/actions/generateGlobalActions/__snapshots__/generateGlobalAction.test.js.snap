// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateGlobalAction happy path 1`] = `
Object {
  "type": "@@rest-easy/@global/@reset/RESET_ALL",
}
`;

exports[`generateGlobalAction no args 1`] = `
Object {
  "type": undefined,
}
`;
